crypto/src/lib.rs

```diff
@@ -119,31 +120,54 @@ pub enum KeyGenOption<K = PrivateKey> {
     Random,
     /// Use seed
     UseSeed(Vec<u8>),
-    /// Derive from private key
+    /// Derive from a private key
     FromPrivateKey(K),
 }
 
+mod config_algorithm_state {
+    use crate::Algorithm;
+
+    #[derive(Copy, Clone, Debug)]
+    pub struct Default;
+
+    impl From<Default> for Algorithm {
+        fn from(_: Default) -> Self {
+            use std::default::Default;
+            Self::default()
+        }
+    }
+
+    #[derive(Copy, Clone, Debug)]
+    pub struct Set(pub Algorithm);
```

Comment:
```suggestion
    pub struct Custom(pub Algorithm);
```

maybe?

Reply:
I believe I can help you work around FFI macros.

Reply:
There was a problem that `KeyPair::generate_with_configuration` started to have a generic parameter to accept `KeyGenConfig<A>`, and FFI macros said generics aren't supported.

And this piece errored with a missing generic argument. Didn't succeed with adding one.

https://github.com/hyperledger/iroha/blob/e79f617f5edc53621e4561f057df92b425beeaca/crypto/src/lib.rs#L867-L873

Reply:
| see, we can't have generic objects in FFI.

in this case, you should have specified all concrete types like this:
```rs
 iroha_ffi::handles! { 
     KeyPairGenConfig<Default>, 
     KeyPairGenConfig<Set>,
     PublicKey, 
     PrivateKey, 
     KeyPair, 
     Signature, 
 }
 ```

Reply:
but then I'm not sure what would have happened with generic functions. I think there was only one potentially problematic `generate_with_configuration`. If you want to go back to this, tell me and we can work something out

Reply:
It's fine, I removed `with_algorithm` method and this problem is gone.

Reply:
I think you found a very good alternative